<!DOCTYPE html>
<html lang="en">

<head>
    <meta charset="UTF-8">
    <title>Metagenomic Visualization</title>
    <link rel="stylesheet" href="styles.css" />
    <link href="https://fonts.googleapis.com/css?family=Work+Sans:400,700&display=swap" rel="stylesheet">
    <script src="https://unpkg.com/d3-simple-slider"></script>
    <script src="https://d3js.org/d3.v5.js"></script>
    <script src="js/StackedBar.js" async></script>
    <script src="js/sunburst.js" async></script>
    <script src="js/Phylo.js" async></script>
    <script src="js/ViolinPlot.js" async></script>
    <script src="js/newHeatMap.js" async></script>
    <script src="js/script.js" async></script>
</head>

<body>
    <div class="header">
        <header>
            <h1>Visualization for Metagenomic Data</h1>
            <h2>Black-eared Mice Gut Microbiome</h2>
            <div>Authors:  LeAnn Lindsey, Kimberly Truong, Lourdes Valdez</div>
        </header>
    </div>
    <div class="wrapper">
        <div class="innerWrapper"> 
            <h3>Phylogenetic Tree & Sunburst Plot</h3>
            <div class="phylo_text">
                <p>This phylogenetic tree shows the species found in all host samples.  Click on the nodes to expand or collapse each branch. You can also zoom in or out and scroll for enhanced visualization.</p>
                <p>This sunburst plot shows the species found in one host sample.  You can choose which sample to look at by clicking on the sample name in the x-axis labels of the Stacked Bar Chart.</p>
                <form action="https://youtu.be/HRD0133kAhM" method="get" target="_blank">
                    <button type="submit">Click Here for an Instructional Tutorial</button>
                 </form>
            </div>
            <div id = "chooseview-radio" class="radio-div">
                <h3>Choose View:</h3>
                    <input type="radio" name="phylosun" value="Phylo" checked>View Phylogenetic Tree</br>
                    <input type="radio" name="phylosun" value="Sunburst">View Sunburst Plot</br>
                <h3 id='sunburstLabel'></h3>
            </div>
            <div id="Phylo" class="view"> </div> 
        </div>
        
        <div class="innerWrapperRight"> 
            <h3>Species Abundance per Sample</h3>
            <div class="stack_text">
            <p>The stacked bar chart shows the taxon abundance within each group of experimental samples.</p>
            <p>Hover over each stacked bar to see the taxon it corresponds to and the percentage of abundance within the chosen experimental group.  You can filter the taxon shown by clicking on the category to remove in the legend.</p>
            </div>
            <div id="stacked-barchart" class="view">
                <h3 id='stackedLabel'>Level: Phylum</h3>
            </div>
        </div>
        <div class="innerWrapperBottomLeft">
            <h3>Gene Abundance by Sample</h3>
                <div class="heat_text">
                <p>The heatmap shows the abundance of each gene by sample.  This is a selection of differentially expressed genes.  Data was processed with edgeR and a p-value of 0.001 was used as a cutoff.</p>
                <p>You can also choose to view the gene pathways in the heatmap.  All of the gene pathways exported by Humann3 are listed below.</p>
                <p>Hover over each square for information on abundance and gene function. Scroll down to see more genes.</p>
                </div>
                <div id = "choosefile-radio" class="radio-div">
                    <h3>Choose Data Input for Heatmap</h3>
                        <input type="radio" name="heatmapFile" value="geneFamily" checked>View Genes in Heatmap</br>
                        <input type="radio" name="heatmapFile" value="pathways">View Pathways in Heatmap</br>
                </div>
            <div id="heatmap" class="view"> 
                   
            </div>
        </div>
        <div class="innerWrapperBottomRight">
            <h3>Gene Detail</h3>
            <div class="violin_text">
                <p>The violin plot shows the gene abundance by sample for a selected gene separated by experimental condition.</p>
                <p>Click on a gene in the heatmap to update.</p>
            </div>
            <div id="violinLables" class="view">  
                <h3 id='violinLabelGene'>Gene: </h3>
                <h3 id='violinLabelSpecies'>Species: </h3>
                <br></br>
                <input type="button" id="saveButton" value="Save this Gene for Further Analysis" >
            </div>
            <div id="violinplot" class="violinPlotClass"> 
            </div> 
            <div id="GeneToSaveLabel" class="view">  
                <h3 id='Saved Genes'>Saved Genes: </h3>
                <div class="saved_genes_text">
                    <p>Click on the button above to save a gene in a list below.</p>
                    <p>Click on a gene in the list to see gene ontology information from UniProt.</p>
                </div>
            </div>

        </div>
        <div class="innerWrapperBottomSection">
            <h3>Acknowledgements:</h3>
            <div class="acknowlege_text">
                <p>All data was collected by Rodolfo Martinez-Mota in Denise Dearings Lab, Dept of Biology, University of Utah.</p>
                <p>Data was processed using Humann3, a metagenomic processing software created by The Huttenhower Lab, Dept of Biostatistics, Harvard T.H. Chan School of Public Health.</p>
                <p>Gene Ontology information on link out is provided by UniProt.org.</p>
                <p>Pathway information on link out is provided by metacyc.org.</p>
                <p>Special thanks to Youjia Zhou, our TA Mentor.</p>
            </div>
        </div>
    </div>
</body>

</html>

